app/lib/tokumei/close_connection.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Tokumei.CloseConnection do
+  @moduledoc """
+  If we respond to a request with Connection: close, we should send a response
+  back with Connection: close and then close it ourselves.
+
+  This works around buggy clients that send Connection: close and never close the
+  connection themselves.
+  """
+
+  defmacro __using__(_opts) do
+    quote do
+      @before_compile unquote(__MODULE__)
+    end
+  end
+  defmacro __before_compile__(_env) do
+    quote location: :keep do
+      defoverridable [handle_request: 2]
+
+      def handle_request(request, env) do
+        response = super(request, env)
+
+        response_headers = case request.headers do
+          %{"connection" => "close"} -> [{"connection", "close"} | response.headers]
```

Comment:
are request headers really a map? If so I have forgotten that change.

second should not add connection close if it is already in the response headers